niworkflows/interfaces/nibabel.py

```diff
@@ -72,20 +79,115 @@ class Binarize(SimpleInterface):
     def _run_interface(self, runtime):
         img = nb.load(self.inputs.in_file)
 
-        self._results['out_file'] = fname_presuffix(
-            self.inputs.in_file, suffix='_masked', newpath=runtime.cwd)
-        self._results['out_mask'] = fname_presuffix(
-            self.inputs.in_file, suffix='_mask', newpath=runtime.cwd)
+        self._results["out_file"] = fname_presuffix(
+            self.inputs.in_file, suffix="_masked", newpath=runtime.cwd
+        )
+        self._results["out_mask"] = fname_presuffix(
+            self.inputs.in_file, suffix="_mask", newpath=runtime.cwd
+        )
 
         data = img.get_fdata()
         mask = data > self.inputs.thresh_low
         data[~mask] = 0.0
         masked = img.__class__(data, img.affine, img.header)
-        masked.to_filename(self._results['out_file'])
+        masked.to_filename(self._results["out_file"])
 
-        img.header.set_data_dtype('uint8')
-        maskimg = img.__class__(mask.astype('uint8'), img.affine,
-                                img.header)
-        maskimg.to_filename(self._results['out_mask'])
+        img.header.set_data_dtype("uint8")
+        maskimg = img.__class__(mask.astype("uint8"), img.affine, img.header)
+        maskimg.to_filename(self._results["out_mask"])
 
         return runtime
+
+
+class _SplitSeriesInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc="input 4d image")
+    allow_3D = traits.Bool(
+        False, usedefault=True, desc="do not fail if a 3D volume is passed in"
+    )
+
+
+class _SplitSeriesOutputSpec(TraitedSpec):
+    out_files = OutputMultiObject(File(exists=True), desc="output list of 3d images")
+
+
+class SplitSeries(SimpleInterface):
+    """Split a 4D dataset along the last dimension into a series of 3D volumes."""
+
+    input_spec = _SplitSeriesInputSpec
+    output_spec = _SplitSeriesOutputSpec
+
+    def _run_interface(self, runtime):
+        filenii = nb.squeeze_image(nb.load(self.inputs.in_file))
+        filenii = filenii.__class__(
+            np.squeeze(filenii.dataobj), filenii.affine, filenii.header
+        )
+        ndim = filenii.dataobj.ndim
+        if ndim != 4:
+            if self.inputs.allow_3D and ndim == 3:
```

Comment:
This is odd, as the above will coerce a valid 4D `(x, y, z, 1)` image to 3D `(x, y, z)`, requiring you to then `allow_3D`.

Reply:
why is that odd? It is indeed a 3D volume, right?

Reply:
for instance, there are a fair number of T1w images in OpenNeuro with (x, y, z, 1) dimensions.

Reply:
It is a 4D series. You should be able to split it into one 3D volume without special casing it.

Reply:
I guess the above snippet you wrote would make this particular use-case a standard one?

Reply:
I would say checking for 3D volumes should happen before reshaping the image.